File: java/sui-backend-app/src/main/java/ru/mos/sms/rest/payload/LoginRequest.java
package ru.mos.sms.rest.payload;

import lombok.Data;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;

import javax.validation.constraints.NotBlank;

@Data
public class LoginRequest {
    @NotBlank
    private String usernameOrEmail;
    @NotBlank
    private String password;

    public UsernamePasswordAuthenticationToken toAuthenticationToken() {
        return new UsernamePasswordAuthenticationToken(this.usernameOrEmail, this.password);
    }
}

File: java/sui-security/src/main/java/ru/sui/suisecurity/model/LoginRequest.kt
package ru.sui.suisecurity.model

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import javax.validation.constraints.NotBlank


class LoginRequest(
        @NotBlank
        val usernameOrEmail: String,
        @NotBlank
        val password: String
) {

    fun toAuthenticationToken() = UsernamePasswordAuthenticationToken(usernameOrEmail, password)

}
